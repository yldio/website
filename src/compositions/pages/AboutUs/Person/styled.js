import styled from 'styled-components';
import { spacing } from 'styles/spacing';
import { Copy } from 'components/Typography';

export const Img = styled.img`
  border-radius: 50%;
  width: 180px;
  width: 180px;
`;

export const Title = styled(Copy)`
  margin-top: ${spacing(-2)};
`;
