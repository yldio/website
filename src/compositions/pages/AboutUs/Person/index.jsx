import React from 'react';
import PropTypes from 'prop-types';
import Link from 'components/Link';
import { Copy } from 'components/Typography';

import { Img, Title } from './styled';

const Picture = ({ picture, name, title, twitter, linkedin, children }) => (
  <li>
    <Img src={picture} alt={name} width="180" height="180" />
    <Copy bold> {name}</Copy>
    <Title> {title}</Title>
    <Copy> {children}</Copy>
    <Copy>
      <Link href={twitter} target="_blank">
        Twitter
      </Link>
    </Copy>
    <Copy>
      <Link href={linkedin} target="_blank">
        LinkedIn
      </Link>
    </Copy>
  </li>
);

export default Picture;

Picture.propTypes = {
  children: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  picture: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  twitter: PropTypes.string.isRequired,
  linkedin: PropTypes.string.isRequired,
};
