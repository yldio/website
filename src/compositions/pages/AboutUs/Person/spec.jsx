import React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';

import Person from './index';

test('components/Person snapshot', () => {
  const wrapper = mount(
    <Person
      picture="/image.hs"
      name="Nuno Job"
      title="Founder + Director"
      twitter="https://twitter.com/dscape"
      linkedin="https://twitter.com/dscape"
    >
      Nuno is the founder and director at YLD. The previous CCO at Nodejitsu,
      Nuno was responsible for business development and sales in the worldâ€™s
      largest Node.js cloud. Now, he leads the YLD team and inspires all the
      creative minds that make up the company.
    </Person>,
  );
  expect(wrapper.find('styled__Img').prop('src')).toBe('/image.hs');
  expect(wrapper.find('styled__Title p').text()).toBe(' Founder + Director');

  expect(wrapper).toMatchSnapshot();
});
