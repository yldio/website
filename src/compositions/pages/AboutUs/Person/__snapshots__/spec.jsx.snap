// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Person snapshot 1`] = `
.c4 {
  color: #616680;
  font-weight: 500;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c1 {
  white-space: normal;
  color: #637392;
  font-size: 1.25rem;
  font-weight: 600;
  line-height: 1.6em;
  margin-top: 20px;
  margin-bottom: 20px;
}

.c3 {
  white-space: normal;
  color: #637392;
  font-size: 1.25rem;
  font-weight: 400;
  line-height: 1.6em;
  margin-top: 20px;
  margin-bottom: 20px;
}

.c0 {
  border-radius: 50%;
  width: 180px;
  width: 180px;
}

.c2 {
  margin-top: -20px;
}

<Picture
  linkedin="https://twitter.com/dscape"
  name="Nuno Job"
  picture="/image.hs"
  title="Founder + Director"
  twitter="https://twitter.com/dscape"
>
  <li>
    <styled__Img
      alt="Nuno Job"
      height="180"
      src="/image.hs"
      width="180"
    >
      <img
        alt="Nuno Job"
        className="c0"
        height="180"
        src="/image.hs"
        width="180"
      />
    </styled__Img>
    <Typography__base
      bold={true}
    >
      <p
        className="c1"
      >
         
        Nuno Job
      </p>
    </Typography__base>
    <styled__Title>
      <Typography__base
        className="c2"
      >
        <p
          className="c2 c3"
        >
           
          Founder + Director
        </p>
      </Typography__base>
    </styled__Title>
    <Typography__base>
      <p
        className="c3"
      >
         
        Nuno is the founder and director at YLD. The previous CCO at Nodejitsu, Nuno was responsible for business development and sales in the worldâ€™s largest Node.js cloud. Now, he leads the YLD team and inspires all the creative minds that make up the company.
      </p>
    </Typography__base>
    <Typography__base>
      <p
        className="c3"
      >
        <Link
          activeClassName="___active"
          href="https://twitter.com/dscape"
          target="_blank"
        >
          <styled__StyledLink
            href="https://twitter.com/dscape"
            target="_blank"
          >
            <a
              className="c4"
              href="https://twitter.com/dscape"
              target="_blank"
            >
              Twitter
            </a>
          </styled__StyledLink>
        </Link>
      </p>
    </Typography__base>
    <Typography__base>
      <p
        className="c3"
      >
        <Link
          activeClassName="___active"
          href="https://twitter.com/dscape"
          target="_blank"
        >
          <styled__StyledLink
            href="https://twitter.com/dscape"
            target="_blank"
          >
            <a
              className="c4"
              href="https://twitter.com/dscape"
              target="_blank"
            >
              LinkedIn
            </a>
          </styled__StyledLink>
        </Link>
      </p>
    </Typography__base>
  </li>
</Picture>
`;
